mbuild/tests/test_charmm_writer.py

```diff
@@ -828,6 +924,375 @@ def test_charmm_pdb_fix_angle_bond_fix_atoms(
+                    and "b0" in line
+                    and "atoms_types_per_utilized_FF" in line
+                ):
+                    bonds_read = True
+                    bond_types = [
+                        ["D", "G", "340.0", "1.09"],
+                        ["E", "G", "320.0", "1.41"],
+                        ["E", "F", "553.0", "0.945"],
+                        ["A", "C", "999999999999", "1.09"],
+                        ["B", "D", "340.0", "1.09"],
+                        ["A", "A", "999999999999", "1.529"],
+                        ["B", "G", "268.0", "1.529"],
+                    ]
+                    total_bonds_evaluated = []
+                    total_fixed_bonds = []
+                    for j in range(0, 7):
+                        total_bonds_evaluated.append(
+                            out_gomc[i + 1 + j].split("!")[0].split()[0:4]
+                        )
+                        if out_gomc[i + 1 + j].split("!")[0].split()[2:3] == [
+                            "999999999999"
+                        ]:
+                            total_fixed_bonds.append(
+                                out_gomc[i + 1 + j].split("!")[0].split()[0:4]
+                            )
+                    assert total_bonds_evaluated.sort() == bond_types.sort()
+                    assert len(total_fixed_bonds) == 2
+
+                elif (
+                    "!atom_types" in line
+                    and "Ktheta" in line
+                    and "Theta0" in line
+                    and "atoms_types_per_utilized_FF" in line
+                ):
+                    angles_read = True
+                    fixed_angle_types = []
+                    total_angles_evaluated = []
+                    total_fixed_angles = []
+                    for j in range(0, 9):
+                        if len(fixed_angle_types) > 0:
+                            if out_gomc[i + 1 + j].split("!")[0].split()[
+                                0:4
+                            ] == (fixed_angle_types[0] or fixed_angle_types[1]):
+                                total_angles_evaluated.append(
+                                    out_gomc[i + 1 + j]
+                                    .split("!")[0]
+                                    .split()[0:4]
+                                )
+                        if out_gomc[i + 1 + j].split("!")[0].split()[3:4] == [
+                            "999999999999"
+                        ]:
+                            total_fixed_angles.append(
+                                out_gomc[i + 1 + j].split("!")[0].split()[0:4]
+                            )
+                    assert (
+                        fixed_angle_types.sort()
+                        == total_angles_evaluated.sort()
+                    )
+                    assert len(total_fixed_angles) == len(fixed_angle_types)
+
+                else:
+                    pass
+
+        bonds_read == True
+        angles_read == True
```

Comment:
Is an assert statement missing here?

Reply:
Good catch, it is added now as:

        assert bonds_read == True
        assert angles_read == True